### PR TITLE
Question answer pages

### PR DESCRIPTION
**Pull request by defabdullah**
Add questions to node details and profile(in question activity) pages. Question in profile page is clickable and redirected to node details page. Node details page has ask question forum and answer question button for contributors.

### CODE REVIEW

**Comment by Brunettow** [comment]
I reviewed the code. Nice job!
The code works without error however there are some parts that should be edited later:
- Questions that are not answered yet should only be visible to node's contributors.
- Designs can be enhanced.

File: project/FrontEnd/collaborative_science_platform/lib/models/profile_data.dart [hunk 1]
```diff
@@ -28,24 +29,33 @@ class ProfileData {
...
       this.nodes = const [],
-      this.askedQuestionIDs = const [],
-      this.answeredQuestionIDs = const []});
+      this.askedQuestions = const [],
+      this.answeredQuestions = const []});
+
   factory ProfileData.fromJson(Map<String, dynamic> jsonString) {
-    var list = jsonString['nodes'] as List;
-    List<Node> nodes = list.map((e) => Node.fromJson(e)).toList();
+    var nodeList = jsonString['nodes'] as List;
```
Comment (Brunettow) [inline project/FrontEnd/collaborative_science_platform/lib/models/profile_data.dart:44]: New version is butter, nice job.

File: project/FrontEnd/collaborative_science_platform/lib/main.dart [hunk 2]
```diff
@@ -35,42 +36,16 @@ class MyApp extends StatelessWidget {
         ChangeNotifierProvider<NodeProvider>(create: (context) => NodeProvider()),
         ChangeNotifierProvider<UserProvider>(create: (context) => UserProvider()),
         ChangeNotifierProvider<WorkspaceProvider>(create: (context) => WorkspaceProvider()),
+        ChangeNotifierProvider<QuestionAnswerProvider>(
+            create: (context) => QuestionAnswerProvider()),
       ],
-      //    child: MaterialApp(
```
Comment (Brunettow) [inline project/FrontEnd/collaborative_science_platform/lib/main.dart:39]: I think its okey to remove these commented lines however It's better not changing anything out of scope.

**Review by Brunettow — APPROVED** [review]
I reviewed the code. There seems no error or incompatibility issues. Good job.

**Comment by defabdullah** [comment]
> I reviewed the code. Nice job! The code works without error however there are some parts that should be edited later:
> 
> * Questions that are not answered yet should only be visible to node's contributors.
> * Designs can be enhanced.

Questions that are not answered are visible to just contributor of that node. You can track it widget.canAnswer bool value.

**Comment by Brunettow** [comment]
It seems like you have soled the issue about visibility of unanswered questions. Good work!

**Review by Brunettow — APPROVED** [review]
It seems like you have soled the issue about visibility of unanswered questions. Good work!

**Pull request merged**